components/uniset/src/enum_prop_mapping.rs

```diff
@@ -0,0 +1,876 @@
+    m.insert("Yeh_With_Tail", JoiningGroup::YehWithTail);
+    m.insert("Yudh", JoiningGroup::Yudh);
+    m.insert("Yudh_He", JoiningGroup::YudhHe);
+    m.insert("Zain", JoiningGroup::Zain);
+    m.insert("Zhain", JoiningGroup::Zhain);
+    m.get(name).cloned()
+}
+
+fn get_joining_type_enum(name: &str) -> Option<JoiningType> {
+    let mut m: LiteMap<&str, JoiningType> = LiteMap::new();
+    m.insert("C", JoiningType::JoinCausing);
+    m.insert("D", JoiningType::DualJoining);
+    m.insert("L", JoiningType::LeftJoining);
+    m.insert("R", JoiningType::RightJoining);
+    m.insert("T", JoiningType::Transparent);
+    m.insert("U", JoiningType::NonJoining);
+    m.get(name).cloned()
+}
+
+fn get_line_break_enum(name: &str) -> Option<LineBreak> {
+    let mut m: LiteMap<&str, LineBreak> = LiteMap::new();
+    m.insert("AI", LineBreak::Ambiguous);
+    m.insert("AL", LineBreak::Alphabetic);
+    m.insert("B2", LineBreak::BreakBoth);
+    m.insert("BA", LineBreak::BreakAfter);
+    m.insert("BB", LineBreak::BreakBefore);
+    m.insert("BK", LineBreak::MandatoryBreak);
+    m.insert("CB", LineBreak::ContingentBreak);
+    m.insert("CJ", LineBreak::ConditionalJapaneseStarter);
+    m.insert("CL", LineBreak::ClosePunctuation);
+    m.insert("CM", LineBreak::CombiningMark);
+    m.insert("CP", LineBreak::CloseParenthesis);
+    m.insert("CR", LineBreak::CarriageReturn);
+    m.insert("EB", LineBreak::EBase);
+    m.insert("EM", LineBreak::EModifier);
+    m.insert("EX", LineBreak::Exclamation);
+    m.insert("GL", LineBreak::Glue);
+    m.insert("H2", LineBreak::H2);
+    m.insert("H3", LineBreak::H3);
+    m.insert("HL", LineBreak::HebrewLetter);
+    m.insert("HY", LineBreak::Hyphen);
+    m.insert("ID", LineBreak::Ideographic);
+    m.insert("IN", LineBreak::Inseperable);
+    m.insert("IS", LineBreak::InfixNumeric);
+    m.insert("JL", LineBreak::JL);
+    m.insert("JT", LineBreak::JT);
+    m.insert("JV", LineBreak::JV);
+    m.insert("LF", LineBreak::LineFeed);
+    m.insert("NL", LineBreak::NextLine);
+    m.insert("NS", LineBreak::Nonstarter);
+    m.insert("NU", LineBreak::Numeric);
+    m.insert("OP", LineBreak::OpenPunctuation);
+    m.insert("PO", LineBreak::PostfixNumeric);
+    m.insert("PR", LineBreak::PrefixNumeric);
+    m.insert("QU", LineBreak::Quotation);
+    m.insert("RI", LineBreak::RegionalIndicator);
+    m.insert("SA", LineBreak::ComplexContext);
+    m.insert("SG", LineBreak::Surrogate);
+    m.insert("SP", LineBreak::Space);
+    m.insert("SY", LineBreak::BreakSymbols);
+    m.insert("WJ", LineBreak::WordJoiner);
+    m.insert("XX", LineBreak::Unknown);
+    m.insert("ZW", LineBreak::ZWSpace);
+    m.insert("ZWJ", LineBreak::ZWJ);
+    m.get(name).cloned()
+}
+
+fn get_lead_canonical_combining_class_enum(name: &str) -> Option<LeadCanonicalCombiningClass> {
+    let mut m: LiteMap<&str, LeadCanonicalCombiningClass> = LiteMap::new();
+    m.insert("0", LeadCanonicalCombiningClass::NotReordered);
```

Comment:
Optional: For these functions, the string input is mapped to an identical integer output. Consider parsing the string to an int, rather than performing the string match. However, you will still need to ensure that the int is valid, since the ints are sparse.

Reply:
Probably don't want to go to this level of optimization because it applies to only some of the enum variants in only the `ccc`, `lccc`, and `tccc` properties. But the ideas of completeness / consistency of enum discriminant values for identifiers for binary props (#536) and matching enum discriminant values used in ICU4C (#537) still make sense.